prototypes/FormatPage.hs

```diff
@@ -403,15 +497,15 @@ prop_toFromBitmap bits =
   where
     roundTrip = fromBitmap . toBitmap
 
-prop_encodeDecode :: PageLogical -> Property
-prop_encodeDecode p =
-    p === roundTrip p
+prop_encodeDecode :: DiskPageSize -> PageLogical -> Property
+prop_encodeDecode dpgsz p =
+    p === decodePage p'
   where
-    roundTrip = decodePage . encodePage
+    Just p' = encodePage dpgsz p
```

Comment:
We turn of partials for `head` (Wno-x-partials), so that ship has sailed.

Reply:
The great thing about uni patterns in tests is they fail well! You get a source location. Much shorter to write than case + error, and a better error message too.

So I think they're a positively good thing in tests, but of course not ideal in production code.

Reply:
So if one buys that argument, we should turn off the uni patterns warning in all testsuites, not per-module!

Reply:
List's `head` also has `HasCallStack`

Reply:
The point is, it's an assertion. It's not sloppy. "Expect this to return Just, fail the test otherwise". That's part of what the test does, you can't just restructure that away, you can just express that assertion in different ways.

Reply:
It's sloppy assertion. It's not

```
x <- assertJust (somethingWhichReturnsMaybe ...)
```

Make intention clear, don't implcitly use partial pattern matches. Without comment it's impossible to know whether it's intended to stay so, or is it a (implicit) to-do to be addressed.